packages/@storefront/flux-capacitor/src/core/store/middleware.ts

```diff
@@ -69,6 +70,20 @@ export const UNDOABLE_ACTIONS = [
 export namespace Middleware {
   export const validator = validatorMiddleware();
 
+  export function updateHistory(flux: FluxCapacitor): ReduxMiddleware {
+    return (store: Store<any>) => (next) => (action) => {
+      if (action.type === Actions.UPDATE_HISTORY) {
+        const state = store.getState();
+        const { payload: { method, url = Selectors.url(state) } } = action;
+        const filteredState = filterState(state, action.payload);
```

Comment:
Instead of passing the payload in and using it to build up `filterState` so that it gets the right history, should call the `method` after this action has gone through (ie, need to send the method as an action)

Reply:
Probably fine as is